src/profile-logic/profile-data.js

```diff
@@ -360,23 +362,35 @@ export function getTimingsForPath(
     interval,
     isInvertedTree,
     thread,
+    unfilteredThread,
+    sampleIndexOffset,
     categories
   );
 }
 
 /**
  * This function returns the timings for a specific path. The algorithm is
  * adjusted when the call tree is inverted.
+ * Note that the unfilteredThread should be the original thread before any filtering
+ * (by range or other) happens. Also sampleIndexOffset needs to be properly
+ * specified and is the offset to be applied on thread's indexes to access
+ * the same samples in unfilteredThread.
  */
 export function getTimingsForCallNodeIndex(
   needleNodeIndex: IndexIntoCallNodeTable | null,
   { callNodeTable, stackIndexToCallNodeIndex }: CallNodeInfo,
   interval: Milliseconds,
   isInvertedTree: boolean,
   thread: Thread,
+  unfilteredThread: Thread,
+  sampleIndexOffset: number,
   categories: CategoryList
 ): TimingsForPath {
   const { samples, stackTable, funcTable } = thread;
+  const {
+    samples: unfilteredSamples,
+    stackTable: unfilteredStackTable,
+  } = unfilteredThread;
```

Comment:
I'm concerned about someone incorrectly using the unfilteredThread. Normally I prefer destructuring done like this, but I think here it's justified to do it in the arguments, to make it very clear how the unfiltered information is being used.

Reply:
I understand your concern. My only worries is that there are a lot of parameters already and this would make it even more crowded. Also we'd still have `unfilteredSamples` and `unfilteredStackTable`.
Doesn't the name give a good warning already? Maybe I can change the name to something scarier? Any good idea?

Reply:
I'm not sure on a scarier name. The samples and stack table could be individual arguments to keep things a bit cleaner. I'll leave it up to your judgement though.

Reply:
I'll keep it as is for now. I think having individual arguments would bring the same problem, just with samples and stackTable :)